### ISSUE TITLE
jQuery-UI Datepicker: NVDA precedes text with "blank" when using "left/right" arrow key inside textbox

### ISSUE THREAD

**Issue by noobards** [nvaccess/nvda#5767]
This is an extension of [#5765](https://github.com/nvaccess/nvda/issues/5765). Since that thread was closed, I couldn't update it there itself.

So I made an adjustment as to how the datepicker should behave when dates are being navigated. Instead of populating a div, I tried populating the textbox itself to see if that solves the "blank" issue.

The good news is that it works when I use the ctrl+up or ctrl+down arrow key combo. NVDA reads out whatever is populated inside the textbox. However, the same doesn't work when I try with ctrl+left or ctrl+right combo. It reads reads out "blank" before the actual content.

I have setup a fiddle for reference - [JSFIDDLE](https://jsfiddle.net/f118d36d/)

Does this mean there is some conflict happening with the ctrl+left/ctrl+right combo with NVDA?

**Comment by jcsteh** [comment]
You should be able to comment on an issue even once it's closed. Is that 
not the case for you? Note that I didn't "lock" the issue, just closed it.

The issue is that control+leftArrow normally moves to the previous word, 
so NVDA reads the word at the cursor once the cursor moves. That won't 
work for you because the cursor is at the end of the line, so the word 
at the cursor is indeed blank. The reason it works for control+up/down 
is that those commands (normally meant to move to the next/previous 
paragraph) read the paragraph at the cursor. As I noted on #5765, 
relying on editable text behaviour isn't going to work here.

**Comment by noobards** [comment]
Oh, my bad. I wasn't aware that one could comment on closed topics. Sorry about that.

Your explanation makes sense. Thanks for that. The alternative you gave on #5765 was a bit complicated for me and hence I tried out another approach. Guess I'll have to wrap my head around the concept anyhow.

Thanks again for your time.
